sos/report/plugins/openstack_designate.py

```diff
@@ -0,0 +1,95 @@
+        self.add_copy_spec([
+            "/etc/designate/*",
+            self.var_puppet_gen + "/etc/designate/designate.conf",
+            self.var_puppet_gen + "/etc/designate/pools.yaml",
+        ])
+
+        # logs
+        if self.get_option("all_logs"):
+            self.add_copy_spec([
+                "/var/log/designate/*",
+                "/var/log/containers/designate/*",
+            ])
+        else:
+            self.add_copy_spec([
+                "/var/log/designate/*.log",
+                "/var/log/containers/designate/*.log"
+            ])
+
+        # commands
+        self.add_cmd_output('openstack dns service list --all-projects',
+                            subdir='service')
+        self.add_cmd_output('openstack dns quota list --all-projects',
+                            subdir='quota')
+        self.add_cmd_output('openstack zone list --all-projects',
+                            subdir='zone')
+        self.add_cmd_output('openstack ptr record list --all-projects',
+                            subdir='ptr')
+        self.add_cmd_output('openstack zone export list --all-projects',
+                            subdir='export')
+        self.add_cmd_output('openstack zone import list --all-projects',
+                            subdir='import')
+        self.add_cmd_output('openstack zone blacklist list --all-projects',
+                            subdir='blacklist')
+        self.add_cmd_output('openstack zone transfer request list '
+                            '--all-projects', subdir='transfer-request')
+        self.add_cmd_output('openstack zone transfer accept list '
+                            '--all-projects', subdir='transfer-accept')
+        self.add_cmd_output('openstack tld list --all-projects',
+                            subdir='tld')
+        self.add_cmd_output('openstack tsigkey list --all-projects '
+                            '--column name --column algorithm --column scope',
+                            subdir='tsig')
+
+        # get recordsets for each zone
+        cmd = "openstack zone list -f value -c id"
+        ret = self.exec_cmd(cmd)
+        if ret['status'] == 0:
+            for zone in ret['output'].splitlines():
+                zone = zone.split()[0]
+                self.add_cmd_output(
+                    "openstack recordset list --all-projects %s" % zone,
+                    subdir='recordset')
+
+    def postproc(self):
+        protect_keys = [
+            "password", "connection", "transport_url", "admin_password",
+            "ssl_key_password", "ssl_client_key_password",
+            "memcache_secret_key"
+        ]
+        regexp = r"((?m)^\s*(%s)\s*=\s*)(.*)" % "|".join(protect_keys)
+
+        self.do_path_regex_sub("/etc/designate/*", regexp, r"\1*********")
+        self.do_path_regex_sub(
+            self.var_puppet_gen + "/etc/designate/*",
+            regexp, r"\1*********"
+        )
+
+
+class RedHatdesignate(OpenStackDesignate, RedHatPlugin):
```

Comment:
From the other designate PR, there is a request to add Ubuntu support. Let's add that here as well, since you're already carrying the initial commit from the other PR.